product_cost_adjustment/models/product.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021 - Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+from odoo.tools import float_is_zero
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.model
+    def _change_standard_price(self, new_price, account=False):
```

Comment:
AFAICT this feature is out of the box.
Just set a value on the `standard_price` field
See https://github.com/odoo/odoo/blob/f11c807217cccd96fa18fe24c0cbb436bb1830d1/addons/stock_account/models/product.py#L105

PS: we will want to be able to use specific accounts, depending on the Adjustment Type.
For that we need to extend  `get_product_accouns` (https://github.com/odoo/odoo/blob/f11c807217cccd96fa18fe24c0cbb436bb1830d1/addons/stock_account/models/product.py#L85)

No useful arguments to do this, so instead  we will need to set a context key to let the extension code to look for accounts on the right Adjustment Type record. This could be in another PR, I can help with it.